lib/WebDriver/Session.php

```diff
@@ -433,4 +431,77 @@ protected function getElementPath($elementId)
     {
         return sprintf('%s/element/%s', $this->url, $elementId);
     }
+
+    /**
+     * @param array $args
+     *
+     * @return array
+     */
+    private function prepareScriptArguments(array $args)
+    {
+        foreach ($args as $k => $v) {
+            if ($v instanceof Element) {
+                $args[$k] = [Container::WEBDRIVER_ELEMENT_ID => $v->getID()];
+            } elseif (is_array($v)) {
+                $args[$k] = $this->prepareScriptArguments($v);
+            }
+        }
+
+        return $args;
+    }
+
+    /**
+     * @param mixed $data
+     *
+     * @return mixed
+     */
+    private function webDriverElementRecursive($data)
+    {
+        $element = $this->webDriverElement($data);
+        if ($element !== null) {
+            return $element;
+        } elseif (is_array($data)) {
+            foreach ($data as $k => $v) {
+                $data[$k] = $this->webDriverElementRecursive($v);
+            }
+        }
+
+        return $data;
+    }
+
+    /**
+     * Inject a snippet of JavaScript into the page for execution in the context of the currently selected frame. (synchronous)
+     *
+     * @param array{script: string, args: array} $jsonScript
+     *
+     * @return mixed
+     */
+    public function execute(array $jsonScript)
+    {
+        if (isset($jsonScript['args'])) {
+            $jsonScript['args'] = $this->prepareScriptArguments($jsonScript['args']);
+        }
+
+        $result = parent::execute($jsonScript);
```

Comment:
doh. I should have waited until I got home to review and test your changes. There's no parent `execute()` method in my 1.x branch.

Reply:
I was relying on `AbstractWebDriver::__call` magic method -  https://3v4l.org/o4lH7 - to follow the original execution path as possible. But yes, specifying the method seems cleaner.